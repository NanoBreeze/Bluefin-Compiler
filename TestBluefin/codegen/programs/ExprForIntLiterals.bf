/*
Verifies that basic expressions for int literals are okay
*/

void foo(int val) {

	1+2*3; // expect constant folding so no IR generated
	val+5-9*9*val-val/3; // 9*9 will automatically undergo contant folding. Note, for val*9*9, the instructions will be generated
	
	val > 5;
	val >= 5;
	val < 5;
	val <= 5;

	val == 5;
	val != 5;
	-val;

}