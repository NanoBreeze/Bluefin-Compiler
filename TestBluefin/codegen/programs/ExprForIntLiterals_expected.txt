; ModuleID = 'ExprForIntLiterals.bf'
source_filename = "ExprForIntLiterals.bf"

define void @foo(i32 %val) {
entry:
  %addtmp = add i32 %val, 5
  %multmp = mul i32 81, %val
  %subtmp = sub i32 %addtmp, %multmp
  %divtmp = sdiv i32 %val, 3
  %subtmp1 = sub i32 %subtmp, %divtmp
  %cmpSGTtmp = icmp sgt i32 %val, 5
  %cmpSGEtmp = icmp sge i32 %val, 5
  %cmpSLTtmp = icmp slt i32 %val, 5
  %cmpSLEtmp = icmp sle i32 %val, 5
  %cmpEQtmp = icmp eq i32 %val, 5
  %cmpNEtmp = icmp ne i32 %val, 5
  %negtmp = sub i32 0, %val
  %addtmp2 = add i32 %val, 3
  %multmp3 = mul i32 %addtmp2, 3
}
